README.md

```diff
@@ -87,6 +82,14 @@ Install CSI Driver:
 
 The snapshot validating webhook is an HTTP callback which responds to [admission requests](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/). It is part of a larger [plan](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook) to tighten validation for volume snapshot objects. This webhook introduces the [ratcheting validation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook#backwards-compatibility) mechanism targeting the tighter validation. The cluster admin or Kubernetes distribution admin should install the webhook alongside the snapshot controllers and CRDs.
 
+Along with the validation webhook, the volume snapshot controller will start [labeling](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook#automatic-labelling-of-invalid-objects) invalid snapshot objects which already existed. This is to enable quick identification of invalid snapshot objects in the system by running:
```

Comment:
clarify this is for v1beta1 objects

Reply:
For v1beta1, there's a check to make sure there is only one source in the snapshot and also snapshotClassName is not empty string.  Otherwise, invalid label will be applied.

For v1, there's a still a check to make sure snapshotClassName is not empty string and apply label accordingly.

I'll add these.

Reply:
> For v1, there's a still a check to make sure snapshotClassName is not empty string and apply label accordingly.

This cannot be validated by the CRD schema?

Reply:
No, this cannot.  Only the oneOf is validated by the CRD schema.  Xiangqian has verified that.

Reply:
does the validation webhook validate it for v1 objects?  theoretically v1 objects should all be legal right?

Actually how does this work if the controller is reading only v1 objects now?

Reply:
Yes, the webhook will still validate v1 objects.

If there are no existing invalid v1beta1 objects, after upgrading to v1, the webhook and schema validation will prevent the user from creating new invalid v1 and v1beta1 objects.

If there are existing invalid v1beta1 objects, the user should install the validation webhook before upgrading to v1 so that those invalid objects will be labeled and can be identified easily and removed before upgrading to v1.

If there are invalid v1beta1 objects and the user didn't install the validation webhook before upgrading to v1, those existing invalid v1beta1 objects will not be labeled (note: labeling is done by the controller, not the validation webhook).  The user can still manually identify them and delete them.

So the recommendation is that before upgrading to v1, the user should install the validation webhook so that all existing invalid objects will be labeled and can be easily identified and deleted.